lib/rspec/matchers/dsl.rb

```diff
@@ -80,11 +77,20 @@ def match(&match_block)
         # @yield [Object] actual the actual value (i.e. the value wrapped by `expect`)
         def match_when_negated(&match_block)
           define_user_override(:does_not_match?, match_block) do |actual|
-            @actual = actual
-            super(*actual_arg_for(match_block))
+            begin
+              @actual = actual
+              RSpec::Support.with_failure_notifier(RAISE_METHOD) do
+                super(*actual_arg_for(match_block))
+              end
+            rescue RSpec::Expectations::ExpectationNotMetError
+              false
```

Comment:
This is correct, no? `does_not_match?` should return false if the expectation included in a `match_when_negated` block fails(?)